taca/utils/undetermined.py

```diff
@@ -0,0 +1,164 @@
+            for barcode in output.split('\n')[:-1]:
+                try:
+                    barcodes[barcode]=barcodes[barcode]+1
+                except KeyError:
+                    barcodes[barcode]=1
+
+        save_index_count(barcodes, run, lane)
+        total=sum(barcodes.values())
+        count, bar = max((v, k) for k, v in barcodes.items())
+        if total * freq_tresh / 100<count:
+            logger.warn("The most frequent barcode of lane {} ({}) found in {} represents {}%, which is over the threshold of {}%".format(lane, bar, fastqfile, count / total * 100, freq_tresh))
+            return False
+        else:
+            return True
+
+
+
+
+def first_qc_check(lane, lb, und_tresh, q30_tresh):
+    """checks wether the percentage of bases over q30 for the sample is above the treshold, and if the amount of undetermined is below the treshold"""
+    d={}
+    for entry in lb.sample_data:
+        if lane == int(entry['Lane']):
+            if entry.get('Sample')=='unknown':
+                if float(entry['% >= Q30bases']) < q30_tresh:
+                    logger.warn("Undetermined indexes of lane {} has a percentage of bases over q30 of {}%, which is below the cutoff of {}% ".format(lane, float(entry['% >= Q30bases']), q30_tresh))
+                    return False
+                d['undet']=int(entry['Clusters'].replace(',',''))
+            else:
+                if float(entry['% >= Q30bases']) < q30_tresh:
+                    logger.warn("Undetermined indexes od lane {} has a percentage of bases over q30 of {}%, which is below the cutoff of {}% ".format(lane, float(entry['% >= Q30bases']), q30_tresh))
+                    return False
+                d['det']=int(entry['Clusters'].replace(',',''))
+
+    if d['undet'] > d['det']+d['undet'] * und_tresh / 100:
+        logger.warn("Lane {} has more than {}% undetermined indexes ({}%)".format(lane, und_tresh,d['undet']/(d['det']+d['undet'])*100))
+        return False
+
+    return True
+
+
+
+
+def get_path_per_lane(run, ss):
+    d={}
+    for l in ss.data:
+        d[int(l['Lane'])]=os.path.join(run, dmux_folder, l['Project'], l['SampleID'])
+
+    return d
+def get_barcode_per_lane(ss):
+    d={}
+    for l in ss.data:
+        d[int(l['Lane'])]=l['index']
+
+    return d
+
+
+def is_unpooled_lane(ss, lane):
+    count=0
+    for l in ss.data:
+        if int(l['Lane']) == lane:
+            count+=1
+    return count==1
+
+def is_unpooled_run(ss):
+    ar=[]
+    for l in ss.data:
+        ar.append(l['Lane'])
+    return len(ar)==len(set(ar))
+
+        
+
+if __name__=="__main__":
+    import sys
+
+    mainlog = logging.getLogger(__name__)
+    mainlog.setLevel(level=logging.INFO)
+    mfh = logging.StreamHandler(sys.stderr)
+    mft = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+    mfh.setFormatter(mft)
+    mainlog.addHandler(mfh)
+        
+    check_undetermined_status("/srv/illumina/HiSeq_X_data/150424_ST-E00214_0031_BH2WY7CCXX")
```

Comment:
do you need to put the whole flowcell? can't you prepare a trimmed version? at least you could test the parsers.

Never minds if its too much of a hassle, lots of things to do right now